src/data-access/mutations/useAddReactionsComment.ts

```diff
@@ -112,23 +118,35 @@ export const useAddReaction = () => {
         }
         allPosts[postIndex] = updatedPost
         queryClient.setQueryData([QueryKeys.POSTS], allPosts)
-      }
-    },
-    onSuccess: (payload) => {
-      queryClient.setQueryData<FeedPost[]>([QueryKeys.POSTS], (data) => {
-        if (!data) throw new Error('No posts found!')
-        const allPosts = data
-        const postIndex = allPosts?.findIndex((post) => post.id === payload.post.id)
-        if (postIndex !== -1 && allPosts) {
-          const filtered = payload.post.reactions.filter((reaction) => reaction.users.length > 0)
-          payload.post.reactions = filtered
-          allPosts[postIndex] = payload.post
+      } else {
+        const reactionToModify = allPostReactions.find((a) => a.type === reaction.type)
+        if (!reactionToModify) return
+
+        const filteredUsers = reactionToModify?.users.filter((a) => a !== user?.id)
+
+        const updatedReaction = { ...reactionToModify, users: filteredUsers }
+
+        const indexOfUpdatedReaction = allPosts[postIndex].reactions.findIndex(
+          (a) => a.type === reaction.type
+        )
+        const postReactionsWithoutUpdatedOne = allPosts[postIndex].reactions.filter(
+          (a) => a.type !== reaction.type
+        )
+
+        const updatedPost: FeedPost = {
+          ...allPosts[postIndex],
+          reactions: insert(
+            postReactionsWithoutUpdatedOne,
+            indexOfUpdatedReaction,
```

Comment:
what do you think about this?
allReactions[indexOfUpdatedReaction] = updatedReaction
instead of insert fn

Reply:
allPostReactions[indexOfUpdatedReaction] = updatedReaction
updatedPost = {...allPosts[postIndex], reactions: allPostReactions}

i didn't check, just asking, similar solution for replacing a post is used here, then we wouldn't need insert fn
allPosts[postIndex] = updatedPost